src/include/cephfs/libcephfs.h

```diff
@@ -119,16 +119,24 @@ struct ceph_snapdiff_entry_t {
 };
 
 struct ceph_ll_io_info {
-  void (*callback) (struct ceph_ll_io_info *cb_info);
-  void *priv; // private for caller
-  struct Fh *fh;
-  const struct iovec *iov;
-  int iovcnt;
-  int64_t off;
-  int64_t result;
-  bool write;
-  bool fsync;
-  bool syncdataonly;
+	void (*callback) (struct ceph_ll_io_info *cb_info);
+	void *priv; // private for caller
+	struct Fh *fh;
+	struct iovec *iov;
```

Comment:
Is it really necessary to remove the `const` qualifier ? I don't think it will be critical, but it may cause issues in code compiled when the `const` qualifier was present. The compiler optimizations could be incorrect if the iovec is really modified.

Reply:
Well this is original change by @ffilz, who authored this part of the code while implementing zero copy (removing the `const`).
When handling non zerocopy request, the result is copied to `iov`:
https://github.com/ceph/ceph/blob/e9c5000a49b6b4a3eff273c2cdfbaf8af65a8d87/src/libcephfs.cc#L2415

Reply:
The contents of the buffers are not the problem. The problem is modifying the iovec vectors themselves. It seems they can be modified for reads with zerocopy enabled. If that's the only case, then I guess it's fine since for old clients the vectors will remain constant even if not explicitly declared. If that's the only case, I would say we can live with this change.

From the ABI compatibility point of view, I would say that this change is fine. I haven't reviewed the datails of the implementation of the zerocopy itself, though. I can take a look, but I'm not sufficiently familiarized with some parts of the code to do a proper review.

Reply:
Apparently I can't dismiss my own review requesting changes. If someone can, please dismiss it. The changes I requested are already addressed now.